### PR TITLE
Revamp our negative binomial explanation (#5300)

### PR DESCRIPTION
**Pull request by herve-91**
**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [ ] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

File: pymc/distributions/discrete.py [hunk 1]
```diff
@@ -692,15 +692,24 @@ def NegBinom(a, m, x):
 
     .. math::
 
-        \mu &= \frac{n(1-p)}{p} \\
-        \alpha &= n
+        p &= \frac{\alpha}{\mu + \alpha} \\
+        n &= \alpha
 
+    If it is parametrized in terms of n and p, the negative binomial  describes the probability to have x failures 
+    before the n-th success, given the probability p of success in each trial. and its pmf is
```
Comment (michaelosthege) [inline pymc/distributions/discrete.py]:
```suggestion
    If it is parametrized in terms of n and p, the negative binomial describes the probability to have x failures
    before the n-th success, given the probability p of success in each trial. Its pmf is
```

**Review by michaelosthege — COMMENTED** [review]
@ColCarroll does this look good from the math side?

**Comment by ericmjl** [comment]
Having used the NB distribution in multiple places, I can confirm that the explanation is correct. Though, I'm not @ColCarroll and not a mathematician either, I will defer to @ColCarroll!

File: pymc/distributions/discrete.py [hunk 2]
```diff
@@ -692,15 +692,24 @@ def NegBinom(a, m, x):
...
+           
     Parameters
     ----------
-    mu: float
-        Poission distribution parameter (mu > 0).
     alpha: float
-        Gamma distribution parameter (alpha > 0).
+        Gamma distribution shape parameter (alpha > 0).
+    mu: float
+        Gamma distribution mean (mu > 0).
```
Comment (michaelosthege) [inline pymc/distributions/discrete.py]:
```suggestion
        Gamma distribution mean (mu > 0).
```

File: pymc/distributions/discrete.py [hunk 3]
```diff
@@ -692,15 +692,24 @@ def NegBinom(a, m, x):
...
 
+    If it is parametrized in terms of n and p, the negative binomial  describes the probability to have x failures 
+    before the n-th success, given the probability p of success in each trial. and its pmf is
+ 
+    .. math::
+ 
+        f(x \mid n, p) =
+           \binom{x + n - 1}{x}
+           (p)^n (1 - p)^x
+
```
Comment (michaelosthege) [inline pymc/distributions/discrete.py]:
```suggestion

    .. math::

        f(x \mid n, p) =
           \binom{x + n - 1}{x}
           (p)^n (1 - p)^x

```

**Review by michaelosthege — CHANGES_REQUESTED** [review]

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**